core/src/network.rs

```diff
@@ -584,6 +584,33 @@ pub enum DialError<THandler> {
     },
 }
 
+impl<THandler> fmt::Debug for DialError<THandler> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            DialError::ConnectionLimit { limit, handler: _ } => f
+                .debug_struct("DialError::ConnectionLimit")
+                .field("limit", limit)
+                .finish(),
+            DialError::LocalPeerId { handler: _ } => {
+                f.debug_struct("DialError::LocalPeerId").finish()
+            }
+            DialError::InvalidPeerId {
+                handler: _,
+                multihash,
+            } => f
+                .debug_struct("DialError::InvalidPeerId")
+                .field("multihash", multihash)
+                .finish(),
+        }
+    }
+}
+
+impl<THandler> fmt::Display for DialError<THandler> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
```

Comment:
The `Display` implementation of an error should be a nice string IMO. For most other errors in the codebase, we match on `self` here and return something expressive.

Reply:
I've implemented a prettier display implementation. The output for ConnectionLimit and LocalPeerId are inspired by the docs, but there was no description for InvalidPeerId, so I wrote something myself there, I hope that's okay, but would be happy with suggestions for something better.

Reply:
If PeerId map one to one to multihashes, it might be cleaner to say The dialing attempt was rejected because the PeerId was invalid" instead of what I wrote now. Let me know if you think that's better.